jws/jws.go

```diff
@@ -12,34 +12,34 @@ import (
 	"github.com/tbd54566975/web5-go/dids/didcore"
 )
 
-func Decode(jws string) (Decoded, error) {
+func Decode[T any](jws string) (Decoded[T], error) {
 	parts := strings.Split(jws, ".")
 	if len(parts) != 3 {
-		return Decoded{}, fmt.Errorf("malformed JWS. Expected 3 parts, got %d", len(parts))
+		return Decoded[T]{}, fmt.Errorf("malformed JWS. Expected 3 parts, got %d", len(parts))
```

Comment:
Kind of annoying we can't have default the generic type to something like `any` or `interface{}`